server/src/main/java/org/apache/iotdb/db/metadata/MTree.java

```diff
@@ -695,6 +708,38 @@ private int getCount(MNode node, String[] nodes, int idx) throws MetadataExcepti
     }
   }
 
+  /**
+   * Traverse the MTree to get the count of devices.
+   */
+  private int getDevicesCount(MNode node, String[] nodes, int idx) throws MetadataException {
+    String nodeReg = MetaUtils.getNodeRegByIdx(idx, nodes);
+    int cnt = 0;
+    if (!(PATH_WILDCARD).equals(nodeReg)) {
+      if (node.hasChild(nodeReg)) {
+        if (node.getChild(nodeReg) instanceof MeasurementMNode && idx >= nodes.length) {
+          cnt++;
+        } else {
+          cnt += getDevicesCount(node.getChild(nodeReg), nodes, idx + 1);
+        }
+      }
+    } else {
+      boolean deviceAdded = false;
+      for (MNode child : node.getChildren().values()) {
+        if (child instanceof MeasurementMNode ) {
+          if( !deviceAdded){
+            cnt++;
+            deviceAdded = true;
+          }else{
```

Comment:
code format
}else{- > } else {
if( !deviceAdded){. -> if( !deviceAdded) {

please use google style format: http://iotdb.apache.org/zh/Development/ContributeGuide.html

Reply:
i got this